javalib/src/main/scala/java/util/concurrent/atomic/AtomicReferenceArray.scala

```diff
@@ -0,0 +1,43 @@
+package java.util.concurrent.atomic
+
+class AtomicReferenceArray[E <: AnyRef](
+    length: Int) extends Serializable {
+
+  def this(array: Array[E]) = {
+    this(array.size)
+    System.arraycopy(array, 0, inner, 0, length)
+  }
+
+  private val inner: Array[AnyRef] = new Array[AnyRef](length)
```

Comment:
It's in the earlier discussions. For `Array[E]` you need a `ClassTag` to be able to create it. So `Array[AnyRef]` is the next best thing.